src/evaluators/TryStatement.js

```diff
@@ -174,14 +201,14 @@ export default function(ast: BabelNodeTryStatement, strictCode: boolean, env: Le
     if (alternate instanceof PossiblyNormalCompletion || alternate instanceof JoinedAbruptCompletions) {
       alternateEffects = composeNestedThrowEffectsWithHandler(alternate, priorEffects);
     } else {
-      alternateEffects = realm.evaluateForEffects(
+      alternateEffects = evaluateForEffectsWithPriorEffects(
+        realm,
+        priorEffects,
         () => {
-          for (let priorEffect of priorEffects) realm.applyEffects(priorEffect);
           invariant(ast.finalizer);
           return env.evaluateCompletionDeref(ast.finalizer, strictCode);
         },
-        undefined,
-        "composeNestedThrowEffectsWithHandler"
+        "composeNestedThrowEffectsWithHandler/2"
```

Comment:
Let's fix this to be `composeNestedEffectsWithFinalizer/2`. Looks like bad original copypasta.